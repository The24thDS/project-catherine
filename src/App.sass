.App
  --margin-size: 100px
  margin: 0 var(--margin-size)
  width: calc( 100vw - var(--margin-size)*2 )

@media only screen and (max-width: 1000px)
  .App
    --margin-size: 0px

@media only screen and (max-width: 1100px)
  .App
    --margin-size: 50px

@media only screen and (orientation: portrait)
  .App
    --margin-size: 0px
