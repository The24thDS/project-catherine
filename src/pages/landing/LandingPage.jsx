import React from "react";
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiTitle,
  EuiIcon
} from "@elastic/eui";

import "./landingPage.sass";
import blackLogo from "../../assets/logo-black.svg";

export default function LandingPage() {
  document.title = "Project Catherine | Landing";
  return (
    <EuiFlexGroup
      className="landing"
      alignItems="center"
      gutterSize="none"
      responsive={true}
      wrap={true}
    >
      <EuiFlexItem className="landing--information" component="section">
        <EuiText>
          <EuiTitle textTransform="uppercase">
            <h1>Project Catherine</h1>
          </EuiTitle>
          <EuiIcon className="logo" size="original" type={blackLogo} />
          <article>
            <p>
              But I must explain to you how all this mistaken idea of denouncing
              pleasure and praising pain was born and I will give you a complete
              account of the system, and expound the actual teachings of the
              great explorer of the truth, the master-builder of human
              happiness. No one rejects, dislikes, or avoids pleasure itself,
              because it is pleasure, but because those who do not know how to
              pursue pleasure rationally encounter consequences that are
              extremely painful. Nor again is there anyone who loves or pursues
              or desires to obtain pain of itself, because it is pain, but
              because occasionally circumstances occur in which toil and pain
              can procure him some great pleasure. To take a trivial example,
              which of us ever undertakes laborious physical exercise, except to
              obtain some advantage from it? But who has any right to find fault
              with a man who chooses to enjoy a pleasure that has no annoying
              consequences, or one who avoids a pain that
            </p>
          </article>
        </EuiText>
      </EuiFlexItem>
      <EuiFlexItem className="landing--forms" component="section">
        login/signup
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
