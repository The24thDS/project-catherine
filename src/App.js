import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import PropTypes from "prop-types";

import "./App.sass";

import { setLoggedIn } from "./redux/user/user.actions";
import { selectLoggedIn } from "./redux/user/user.selectors";
import LandingPage from "./pages/landing/LandingPage";
import Activation from "./pages/activation/Activation";
import ServerRequest from "./utils/ServerRequest";

class App extends React.Component {
  static propTypes = {
    loggedIn: PropTypes.bool.isRequired,
    setLoggedIn: PropTypes.func.isRequired,
  };

  componentDidMount() {
    const token =
      window.localStorage.getItem("token") ||
      window.sessionStorage.getItem("token");
    const checkToken = async (token) => {
      const req = new ServerRequest(
        "/auth/verifyToken",
        "POST",
        {
          "Content-Type": "application/json",
        },
        {
          input: token,
        }
      );
      const response = await req.send();
      const data = await response.json();
      return data.success;
    };

    if (token !== null) {
      checkToken(token).then((valid) => {
        if (valid) {
          this.props.setLoggedIn(true);
        } else {
          window.localStorage.removeItem("token");
          window.sessionStorage.removeItem("token");
        }
      });
    }
  }

  render() {
    return (
      <div className="App">
        <Switch>
          <Route
            exact
            path="/"
            component={(props) =>
              this.props.loggedIn ? (
                <Redirect to="/feed" />
              ) : (
                  <LandingPage {...props} />
                )
            }
          />
          <Route
            exact
            path="/activation/:token"
            component={(props) =>
              this.props.loggedIn ? (
                <Redirect to="/feed" />
              ) : (
                  <Activation {...props} />
                )
            }
          />
          {/* this route will render if all of the above routes don't */}
          <Route>404 - Not Found</Route>
        </Switch>
      </div>
    );
  }
}

// this function will select from the store and map the results to props
// propName: selectorName
const mapStateToProps = createStructuredSelector({
  loggedIn: selectLoggedIn,
});

// this function will map boundActionCreators to props
const mapDispatchToProps = (dispatch) => ({
  setLoggedIn: (loggedIn) => dispatch(setLoggedIn(loggedIn)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
