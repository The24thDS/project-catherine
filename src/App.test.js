import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";

import App from "./App";
import LandingPage from "./pages/landing/LandingPage";

const mockStore = configureStore([]);

describe("App", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      user: {
        currentUser: {
          name: "Catherine",
          id: 5
        }
      }
    });
  });

  test("snapshot renders", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the Landing page", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(wrapper.find(LandingPage).length).toEqual(1);
  });
});
