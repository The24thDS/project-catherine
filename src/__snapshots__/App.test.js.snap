// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <div
    className="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap landing"
  >
    <section
      className="euiFlexItem euiFlexItem--flexGrow4 landing--information"
    >
      <h1
        className="euiTitle euiTitle--medium euiTitle--uppercase"
      >
        Project Catherine
      </h1>
      <img
        className="euiIcon euiIcon-isLoaded logo"
        src="logo-black.svg"
      />
      <article>
        <p>
          But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but because occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that
        </p>
      </article>
    </section>
    <section
      className="euiFlexItem euiFlexItem--flexGrow1 landing--forms"
    >
      <nav
        className="forms-nav"
      >
        <button
          className="forms-nav-button  forms-nav-button--active"
          onClick={[Function]}
        >
          SIGNUP
        </button>
        <button
          className="forms-nav-button "
          onClick={[Function]}
        >
          LOGIN
        </button>
      </nav>
      <form
        className="euiForm landing-form"
        onSubmit={[Function]}
      >
        <div
          className="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive name-inputs"
        >
          <div
            className="euiFlexItem"
          >
            <div
              className="euiFormRow"
              id="mxngtfx9-row"
            >
              <div
                className="euiFormRow__labelWrapper"
              >
                <label
                  className="euiFormLabel euiFormRow__label"
                  htmlFor="mxngtfx9"
                >
                  First Name
                </label>
              </div>
              <div
                className="euiFormRow__fieldWrapper"
              >
                <div
                  className="euiFormControlLayout"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <input
                      className="euiFieldText"
                      id="mxngtfx9"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="euiFlexItem"
          >
            <div
              className="euiFormRow"
              id="la8020om-row"
            >
              <div
                className="euiFormRow__labelWrapper"
              >
                <label
                  className="euiFormLabel euiFormRow__label"
                  htmlFor="la8020om"
                >
                  Last Name
                </label>
              </div>
              <div
                className="euiFormRow__fieldWrapper"
              >
                <div
                  className="euiFormControlLayout"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <input
                      className="euiFieldText"
                      id="la8020om"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFormRow euiFormRow--fullWidth"
          id="mcqcro2n-row"
          onBlur={[Function]}
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="mcqcro2n"
            >
              Email
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <div
              className="euiFormControlLayout euiFormControlLayout--fullWidth"
            >
              <div
                className="euiFormControlLayout__childrenWrapper"
              >
                <input
                  className="euiFieldText euiFieldText--fullWidth"
                  id="mcqcro2n"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFormRow euiFormRow--fullWidth"
          id="l52nkati-row"
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              className="euiFormLabel euiFormRow__label"
              htmlFor="l52nkati"
            >
              Date of Birth
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <span>
              <span
                className="euiDatePicker euiDatePicker--shadow"
              >
                <div
                  className="euiFormControlLayout euiFormControlLayout--fullWidth"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <div
                      className="react-datepicker-wrapper"
                    >
                      <div
                        className="react-datepicker__input-container"
                      >
                        <input
                          aria-label=""
                          className="euiDatePicker euiFieldText euiFieldText--fullWidth euiFieldText--withIcon"
                          disabled={false}
                          id="l52nkati"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          readOnly={false}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className="euiFormControlLayoutIcons"
                    >
                      <span
                        className="euiFormControlLayoutCustomIcon"
                      >
                        <svg
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          height={16}
                          role="img"
                          style={null}
                          viewBox="0 0 16 16"
                          width={16}
                          xmlns="http://www.w3.org/2000/svg"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </span>
            </span>
          </div>
        </div>
        <div
          className="euiFormRow euiFormRow--fullWidth"
          id="nmjn6025-row"
          onBlur={[Function]}
        >
          <div
            className="euiFormRow__labelWrapper"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="nmjn6025"
            >
              Password
            </label>
          </div>
          <div
            className="euiFormRow__fieldWrapper"
          >
            <span
              className="euiToolTipAnchor full-width"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="euiFormControlLayout euiFormControlLayout--fullWidth"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    className="euiFieldText euiFieldText--fullWidth"
                    minLength="10"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </span>
            <div
              className="euiFormHelpText euiFormRow__text"
              id="nmjn6025-help"
            >
              <div>
                <div
                  className="euiFlexGroup euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
                  style={
                    Object {
                      "height": "20px",
                    }
                  }
                >
                  <div
                    className="euiFlexItem euiFlexItem--flexGrow1"
                    style={
                      Object {
                        "height": "100%",
                      }
                    }
                  >
                    Strength:
                  </div>
                  <div
                    className="euiFlexItem euiFlexItem--flexGrow2"
                  >
                    <progress
                      aria-label="password strength"
                      className="euiProgress euiProgress--native euiProgress--m euiProgress--danger"
                      max={100}
                      value={0}
                    />
                  </div>
                </div>
                <p>
                  Must be at least 10 characters long
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFormRow form-button"
          id="7ppfhqir-row"
        >
          <div
            className="euiFormRow__fieldWrapper"
          >
            <button
              className="euiButton euiButton--primary euiButton--fill"
              disabled={false}
              id="7ppfhqir"
              onBlur={[Function]}
              onFocus={[Function]}
              type="submit"
            >
              <span
                className="euiButton__content"
              >
                <span
                  className="euiButton__text"
                >
                  SIGN UP
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          className="euiText euiText--medium form-message"
        >
          
        </div>
      </form>
    </section>
  </div>
</div>
`;
